lib/handler/connect.c

```diff
@@ -550,8 +552,10 @@ static void tcp_on_proceed(h2o_generator_t *_self, h2o_req_t *req)
         reset_io_timeout(self);
         h2o_socket_read_start(self->sock, tcp_on_read);
     } else {
-        self->read_closed = 1;
-        h2o_send(self->src_req, NULL, 0, H2O_SEND_STATE_FINAL);
+        if (!self->read_closed) {
+            self->read_closed = 1;
+            h2o_send(self->src_req, NULL, 0, H2O_SEND_STATE_FINAL);
+        }
```

Comment:
This change by itself is a no-op, as we have `assert(!self->read_closed)` right above.

Do we think that the above assert is incorrect (i.e., have we seen that assertion failure)?